Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM node:22.12.0-alpine3.21 AS builder
+
+ENV PYTHONUNBUFFERED=1 \
+    PIP_NO_CACHE_DIR=1 \
+    VIRTUAL_ENV=/opt/venv \
+    TERRAFORM_VERSION=1.6.0 \
+    CDKTF_VERSION=latest
+
+RUN apk add --no-cache \
+    python3 python3-dev py3-pip \
+    gcc musl-dev linux-headers libffi-dev openssl-dev build-base \
+    git curl jq unzip bash
+
+RUN curl -fsSL "https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip" -o terraform.zip \
+    && unzip terraform.zip \
+    && mv terraform /usr/local/bin/ \
+    && chmod +x /usr/local/bin/terraform \
+    && rm -f terraform.zip
+
+RUN npm install -g cdktf-cli@$CDKTF_VERSION \
+    && npm cache clean --force
+
+RUN python3 -m venv $VIRTUAL_ENV \
+    && $VIRTUAL_ENV/bin/pip install --upgrade pip \
+    && $VIRTUAL_ENV/bin/pip install psutil requests azure-cli \
```

Comment:
Size of docker images in a final is difference with +1 GB with azure-cli and without 2.8GB vs 1.12GB 


Seems layer is too heavy

with azure-cli
`<missing>      13 minutes ago   COPY /opt/venv /opt/venv # buildkit             1.35GB    buildkit.dockerfile.v0`

without azure-cli

`<missing>      45 hours ago   COPY /opt/venv /opt/venv # buildkit             19.2MB    buildkit.dockerfile.v0`

perhaps there is another ways to install it, need to investigate

Reply:
```
~/>  feature/SETI2C-10 !1 ?6  make apply EMAIL="m.zemlianyi@setuniversity.edu.ua"                                                                                                                                           ✔  3m 40s  14:49:16
Running 'apply' CDKTF Azure stack...
docker run --rm -v /Users/zemliany/.azure.docker.tmp/:/home/cdktf/.azure zemlyanoy/cdktf-python-azure apply m.zemlianyi@setuniversity.edu.ua
Azure CLI credentials are invalid or missing. Triggering az login...
To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code GCJU2VBYP to authenticate.
[
  {
    "cloudName": "AzureCloud",
    "homeTenantId": "a601d367-4a53-48c0-b2a2-ac01e8c21e66",
    "id": "c265face-b525-4252-97e0-82e82734e7ef",
    "isDefault": true,
    "managedByTenants": [],
    "name": "Azure for Students",
    "state": "Enabled",
    "tenantDefaultDomain": "mzemlianyioutlook.onmicrosoft.com",
    "tenantDisplayName": "Default Directory",
    "tenantId": "a601d367-4a53-48c0-b2a2-ac01e8c21e66",
    "user": {
      "name": "m.zemlianyi@outlook.com",
      "type": "user"
    }
  }
]
Azure RBAC Creds is not set or is empty. Setting it now...
Sleep 10...
WARNING: Creating 'Contributor' role assignment under scope '/subscriptions/c265face-b525-4252-97e0-82e82734e7ef'
WARNING: The output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli
Cloning 'cdktf-azure-example'...
Cloning into 'cdktf-azure-example'...
Running 'cdktf get' as it has not been executed yet...
Creating a virtualenv for this project
Pipfile: /app/cdktf-azure-example/Pipfile
Using /usr/local/bin/python3.12.8 to create virtualenv...
created virtual environment CPython3.12.8.final.0-64 in 125ms
  creator
CPython3Posix(dest=/home/cdktf/.local/share/virtualenvs/cdktf-azure-example-pCbo
QNgP, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, via=copy,
app_data_dir=/home/cdktf/.local/share/virtualenv)
    added seed packages: pip==24.3.1
  activators
BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator
,PythonActivator

✔ Successfully created virtual environment!
Virtualenv location: /home/cdktf/.local/share/virtualenvs/cdktf-azure-example-pCboQNgP
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
Installing dependencies from Pipfile.lock (7e9a0f)...
Sleep 10...
Generated python constructs in the output directory: imports
mzemlianyi-azure-stack  Initializing the backend...
mzemlianyi-azure-stack
                        Successfully configured the backend "local"! Terraform will automatically
                        use this backend unless the backend configuration changes.
mzemlianyi-azure-stack  Initializing provider plugins...
mzemlianyi-azure-stack  - Finding hashicorp/random versions matching "3.6.3"...
mzemlianyi-azure-stack  - Finding hashicorp/azurerm versions matching "3.117.0"...
mzemlianyi-azure-stack  - Installing hashicorp/random v3.6.3...
mzemlianyi-azure-stack  - Installed hashicorp/random v3.6.3 (signed by HashiCorp)
mzemlianyi-azure-stack  - Installing hashicorp/azurerm v3.117.0...
mzemlianyi-azure-stack  - Installed hashicorp/azurerm v3.117.0 (signed by HashiCorp)
mzemlianyi-azure-stack  Terraform has created a lock file .terraform.lock.hcl to record the provider
                        selections it made above. Include this file in your version control repository
                        so that Terraform can guarantee to make the same selections by default when
                        you run "terraform init" in the future.

                        Terraform has been successfully initialized!

                        You may now begin working with Terraform. Try running "terraform plan" to see
                        any changes that are required for your infrastructure. All Terraform commands
                        should now work.

                        If you ever set or change modules or backend configuration for Terraform,
                        rerun this command to reinitialize your working directory. If you forget, other
                        commands will detect it and remind you to do so if necessary.
mzemlianyi-azure-stack  - Fetching hashicorp/azurerm 3.117.0 for linux_amd64...
mzemlianyi-azure-stack  - Retrieved hashicorp/azurerm 3.117.0 for linux_amd64 (signed by HashiCorp)
mzemlianyi-azure-stack  - Fetching hashicorp/random 3.6.3 for linux_amd64...
mzemlianyi-azure-stack  - Retrieved hashicorp/random 3.6.3 for linux_amd64 (signed by HashiCorp)
mzemlianyi-azure-stack  - Obtained hashicorp/random checksums for linux_amd64; All checksums for this platform were already tracked in the lock file
                        - Obtained hashicorp/azurerm checksums for linux_amd64; All checksums for this platform were already tracked in the lock file
mzemlianyi-azure-stack  Success! Terraform has validated the lock file and found no need for changes.
mzemlianyi-azure-stack  Terraform used the selected providers to generate the following execution plan.
                        Resource actions are indicated with the following symbols:
                          + create
mzemlianyi-azure-stack  Terraform will perform the following actions:

                          # azurerm_bastion_host.my_bastion (my_bastion) will be created
                          + resource "azurerm_bastion_host" "my_bastion" {
                              + copy_paste_enabled     = true
                              + dns_name               = (known after apply)
                              + file_copy_enabled      = false
                              + id                     = (known after apply)
                              + ip_connect_enabled     = false
                              + kerberos_enabled       = false
                              + location               = "eastus"
                              + name                   = "mzemlianyi-azure-cdktf-bastion"
                              + resource_group_name    = (known after apply)
                              + scale_units            = 2
                              + shareable_link_enabled = false
                              + sku                    = "Standard"
                              + tags                   = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                              + tunneling_enabled      = false

                              + ip_configuration {
                                  + name                 = "ipconfig"
                                  + public_ip_address_id = (known after apply)
                                  + subnet_id            = (known after apply)
                                }
                            }

                          # azurerm_lb.my_lb (my_lb) will be created
                          + resource "azurerm_lb" "my_lb" {
                              + id                   = (known after apply)
                              + location             = "eastus"
                              + name                 = "mzemlianyi-azure-cdktf-lb"
                              + private_ip_address   = (known after apply)
                              + private_ip_addresses = (known after apply)
                              + resource_group_name  = (known after apply)
                              + sku                  = "Standard"
                              + sku_tier             = "Regional"
                              + tags                 = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }

                              + frontend_ip_configuration {
                                  + gateway_load_balancer_frontend_ip_configuration_id = (known after apply)
                                  + id                                                 = (known after apply)
                                  + inbound_nat_rules                                  = (known after apply)
                                  + load_balancer_rules                                = (known after apply)
                                  + name                                               = "frontend-ip"
                                  + outbound_rules                                     = (known after apply)
                                  + private_ip_address                                 = (known after apply)
                                  + private_ip_address_allocation                      = "Dynamic"
                                  + private_ip_address_version                         = (known after apply)
                                  + public_ip_address_id                               = (known after apply)
                                  + public_ip_prefix_id                                = (known after apply)
                                  + subnet_id                                          = (known after apply)
                                }
                            }

                          # azurerm_lb_backend_address_pool.my_lb_pool (my_lb_pool) will be created
mzemlianyi-azure-stack    + resource "azurerm_lb_backend_address_pool" "my_lb_pool" {
                              + backend_ip_configurations = (known after apply)
                              + id                        = (known after apply)
                              + inbound_nat_rules         = (known after apply)
                              + load_balancing_rules      = (known after apply)
                              + loadbalancer_id           = (known after apply)
                              + name                      = "test-pool"
                              + outbound_rules            = (known after apply)
                            }

                          # azurerm_lb_probe.my_lb_probe (my_lb_probe) will be created
                          + resource "azurerm_lb_probe" "my_lb_probe" {
                              + id                  = (known after apply)
                              + interval_in_seconds = 15
                              + load_balancer_rules = (known after apply)
                              + loadbalancer_id     = (known after apply)
                              + name                = "test-probe"
                              + number_of_probes    = 2
                              + port                = 80
                              + probe_threshold     = 1
                              + protocol            = "Tcp"
                            }

                          # azurerm_lb_rule.my_lb_rule (my_lb_rule) will be created
                          + resource "azurerm_lb_rule" "my_lb_rule" {
                              + backend_address_pool_ids       = (known after apply)
                              + backend_port                   = 80
                              + disable_outbound_snat          = true
                              + enable_floating_ip             = false
                              + frontend_ip_configuration_id   = (known after apply)
                              + frontend_ip_configuration_name = "frontend-ip"
                              + frontend_port                  = 80
                              + id                             = (known after apply)
                              + idle_timeout_in_minutes        = (known after apply)
                              + load_distribution              = (known after apply)
                              + loadbalancer_id                = (known after apply)
                              + name                           = "test-rule"
                              + probe_id                       = (known after apply)
                              + protocol                       = "Tcp"
                            }

                          # azurerm_linux_virtual_machine.my_vm_0 (my_vm_0) will be created
                          + resource "azurerm_linux_virtual_machine" "my_vm_0" {
                              + admin_password                                         = (sensitive value)
                              + admin_username                                         = "mzemlianyi"
                              + allow_extension_operations                             = true
                              + bypass_platform_safety_checks_on_user_schedule_enabled = false
                              + computer_name                                          = (known after apply)
                              + disable_password_authentication                        = false
                              + disk_controller_type                                   = (known after apply)
                              + extensions_time_budget                                 = "PT1H30M"
                              + id                                                     = (known after apply)
                              + location                                               = "eastus"
                              + max_bid_price                                          = -1
                              + name                                                   = "mzemlianyi-azure-cdktf-vm-0"
                              + network_interface_ids                                  = (known after apply)
                              + patch_assessment_mode                                  = "ImageDefault"
                              + patch_mode                                             = "ImageDefault"
                              + platform_fault_domain                                  = -1
                              + priority                                               = "Regular"
                              + private_ip_address                                     = (known after apply)
mzemlianyi-azure-stack  + private_ip_addresses                                   = (known after apply)
                              + provision_vm_agent                                     = true
                              + public_ip_address                                      = (known after apply)
                              + public_ip_addresses                                    = (known after apply)
                              + resource_group_name                                    = (known after apply)
                              + size                                                   = "Standard_B1ms"
                              + tags                                                   = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                              + virtual_machine_id                                     = (known after apply)
                              + vm_agent_platform_updates_enabled                      = false

                              + os_disk {
                                  + caching                   = "ReadWrite"
                                  + disk_size_gb              = (known after apply)
                                  + name                      = "mzemlianyi-azure-cdktf-disk-0"
                                  + storage_account_type      = "Standard_LRS"
                                  + write_accelerator_enabled = false
                                }

                              + source_image_reference {
                                  + offer     = "0001-com-ubuntu-server-jammy"
                                  + publisher = "Canonical"
                                  + sku       = "22_04-lts-gen2"
                                  + version   = "latest"
                                }
                            }

                          # azurerm_linux_virtual_machine.my_vm_1 (my_vm_1) will be created
                          + resource "azurerm_linux_virtual_machine" "my_vm_1" {
                              + admin_password                                         = (sensitive value)
                              + admin_username                                         = "mzemlianyi"
                              + allow_extension_operations                             = true
                              + bypass_platform_safety_checks_on_user_schedule_enabled = false
                              + computer_name                                          = (known after apply)
                              + disable_password_authentication                        = false
                              + disk_controller_type                                   = (known after apply)
                              + extensions_time_budget                                 = "PT1H30M"
                              + id                                                     = (known after apply)
                              + location                                               = "eastus"
                              + max_bid_price                                          = -1
                              + name                                                   = "mzemlianyi-azure-cdktf-vm-1"
                              + network_interface_ids                                  = (known after apply)
                              + patch_assessment_mode                                  = "ImageDefault"
                              + patch_mode                                             = "ImageDefault"
                              + platform_fault_domain                                  = -1
                              + priority                                               = "Regular"
                              + private_ip_address                                     = (known after apply)
                              + private_ip_addresses                                   = (known after apply)
                              + provision_vm_agent                                     = true
                              + public_ip_address                                      = (known after apply)
mzemlianyi-azure-stack  + public_ip_addresses                                    = (known after apply)
                              + resource_group_name                                    = (known after apply)
                              + size                                                   = "Standard_B1ms"
                              + tags                                                   = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                              + virtual_machine_id                                     = (known after apply)
                              + vm_agent_platform_updates_enabled                      = false

                              + os_disk {
                                  + caching                   = "ReadWrite"
                                  + disk_size_gb              = (known after apply)
                                  + name                      = "mzemlianyi-azure-cdktf-disk-1"
                                  + storage_account_type      = "Standard_LRS"
                                  + write_accelerator_enabled = false
                                }

                              + source_image_reference {
                                  + offer     = "0001-com-ubuntu-server-jammy"
                                  + publisher = "Canonical"
                                  + sku       = "22_04-lts-gen2"
                                  + version   = "latest"
                                }
                            }

                          # azurerm_linux_virtual_machine.my_vm_2 (my_vm_2) will be created
                          + resource "azurerm_linux_virtual_machine" "my_vm_2" {
                              + admin_password                                         = (sensitive value)
                              + admin_username                                         = "mzemlianyi"
                              + allow_extension_operations                             = true
                              + bypass_platform_safety_checks_on_user_schedule_enabled = false
                              + computer_name                                          = (known after apply)
                              + disable_password_authentication                        = false
                              + disk_controller_type                                   = (known after apply)
                              + extensions_time_budget                                 = "PT1H30M"
                              + id                                                     = (known after apply)
                              + location                                               = "eastus"
                              + max_bid_price                                          = -1
                              + name                                                   = "mzemlianyi-azure-cdktf-vm-2"
                              + network_interface_ids                                  = (known after apply)
                              + patch_assessment_mode                                  = "ImageDefault"
                              + patch_mode                                             = "ImageDefault"
                              + platform_fault_domain                                  = -1
                              + priority                                               = "Regular"
                              + private_ip_address                                     = (known after apply)
                              + private_ip_addresses                                   = (known after apply)
                              + provision_vm_agent                                     = true
                              + public_ip_address                                      = (known after apply)
                              + public_ip_addresses                                    = (known after apply)
                              + resource_group_name                                    = (known after apply)
                              + size                                                   = "Standard_B1ms"
mzemlianyi-azure-stack  + tags                                                   = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                              + virtual_machine_id                                     = (known after apply)
                              + vm_agent_platform_updates_enabled                      = false

                              + os_disk {
                                  + caching                   = "ReadWrite"
                                  + disk_size_gb              = (known after apply)
                                  + name                      = "mzemlianyi-azure-cdktf-disk-2"
                                  + storage_account_type      = "Standard_LRS"
                                  + write_accelerator_enabled = false
                                }

                              + source_image_reference {
                                  + offer     = "0001-com-ubuntu-server-jammy"
                                  + publisher = "Canonical"
                                  + sku       = "22_04-lts-gen2"
                                  + version   = "latest"
                                }
                            }

                          # azurerm_nat_gateway.my_nat_gateway (my_nat_gateway) will be created
                          + resource "azurerm_nat_gateway" "my_nat_gateway" {
                              + id                      = (known after apply)
                              + idle_timeout_in_minutes = 4
                              + location                = "eastus"
                              + name                    = "mzemlianyi-azure-cdktf-nat"
                              + resource_group_name     = (known after apply)
                              + resource_guid           = (known after apply)
                              + sku_name                = "Standard"
                              + tags                    = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                            }

                          # azurerm_nat_gateway_public_ip_association.my_nat_gateway_ip_association (my_nat_gateway_ip_association) will be created
                          + resource "azurerm_nat_gateway_public_ip_association" "my_nat_gateway_ip_association" {
                              + id                   = (known after apply)
                              + nat_gateway_id       = (known after apply)
                              + public_ip_address_id = (known after apply)
                            }

                          # azurerm_network_interface.my_nic_0 (my_nic_0) will be created
                          + resource "azurerm_network_interface" "my_nic_0" {
                              + accelerated_networking_enabled = (known after apply)
                              + applied_dns_servers            = (known after apply)
                              + dns_servers                    = (known after apply)
                              + enable_accelerated_networking  = (known after apply)
                              + enable_ip_forwarding           = (known after apply)
                              + id                             = (known after apply)
                              + internal_domain_name_suffix    = (known after apply)
                              + ip_forwarding_enabled          = (known after apply)
                              + location                       = "eastus"
                              + mac_address                    = (known after apply)
                              + name                           = "mzemlianyi-azure-cdktf-nsg-0"
                              + private_ip_address             = (known after apply)
                              + private_ip_addresses           = (known after apply)
                              + resource_group_name            = (known after apply)
                              + tags                           = {
mzemlianyi-azure-stack  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                              + virtual_machine_id             = (known after apply)

                              + ip_configuration {
                                  + gateway_load_balancer_frontend_ip_configuration_id = (known after apply)
                                  + name                                               = "ipconfig-0"
                                  + primary                                            = true
                                  + private_ip_address                                 = (known after apply)
                                  + private_ip_address_allocation                      = "Dynamic"
                                  + private_ip_address_version                         = "IPv4"
                                  + subnet_id                                          = (known after apply)
                                }
                            }

                          # azurerm_network_interface.my_nic_1 (my_nic_1) will be created
                          + resource "azurerm_network_interface" "my_nic_1" {
                              + accelerated_networking_enabled = (known after apply)
                              + applied_dns_servers            = (known after apply)
                              + dns_servers                    = (known after apply)
                              + enable_accelerated_networking  = (known after apply)
                              + enable_ip_forwarding           = (known after apply)
                              + id                             = (known after apply)
                              + internal_domain_name_suffix    = (known after apply)
                              + ip_forwarding_enabled          = (known after apply)
                              + location                       = "eastus"
                              + mac_address                    = (known after apply)
                              + name                           = "mzemlianyi-azure-cdktf-nsg-1"
                              + private_ip_address             = (known after apply)
                              + private_ip_addresses           = (known after apply)
                              + resource_group_name            = (known after apply)
                              + tags                           = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                              + virtual_machine_id             = (known after apply)

                              + ip_configuration {
                                  + gateway_load_balancer_frontend_ip_configuration_id = (known after apply)
                                  + name                                               = "ipconfig-1"
                                  + primary                                            = true
                                  + private_ip_address                                 = (known after apply)
                                  + private_ip_address_allocation                      = "Dynamic"
                                  + private_ip_address_version                         = "IPv4"
                                  + subnet_id                                          = (known after apply)
                                }
                            }

                          # azurerm_network_interface.my_nic_2 (my_nic_2) will be created
                          + resource "azurerm_network_interface" "my_nic_2" {
                              + accelerated_networking_enabled = (known after apply)
                              + applied_dns_servers            = (known after apply)
                              + dns_servers                    = (known after apply)
                              + enable_accelerated_networking  = (known after apply)
mzemlianyi-azure-stack  + enable_ip_forwarding           = (known after apply)
                              + id                             = (known after apply)
                              + internal_domain_name_suffix    = (known after apply)
                              + ip_forwarding_enabled          = (known after apply)
                              + location                       = "eastus"
                              + mac_address                    = (known after apply)
                              + name                           = "mzemlianyi-azure-cdktf-nsg-2"
                              + private_ip_address             = (known after apply)
                              + private_ip_addresses           = (known after apply)
                              + resource_group_name            = (known after apply)
                              + tags                           = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                              + virtual_machine_id             = (known after apply)

                              + ip_configuration {
                                  + gateway_load_balancer_frontend_ip_configuration_id = (known after apply)
                                  + name                                               = "ipconfig-2"
                                  + primary                                            = true
                                  + private_ip_address                                 = (known after apply)
                                  + private_ip_address_allocation                      = "Dynamic"
                                  + private_ip_address_version                         = "IPv4"
                                  + subnet_id                                          = (known after apply)
                                }
                            }

                          # azurerm_network_interface_backend_address_pool_association.my_nic_lb_pool_0 (my_nic_lb_pool_0) will be created
                          + resource "azurerm_network_interface_backend_address_pool_association" "my_nic_lb_pool_0" {
                              + backend_address_pool_id = (known after apply)
                              + id                      = (known after apply)
                              + ip_configuration_name   = "ipconfig-0"
                              + network_interface_id    = (known after apply)
                            }

                          # azurerm_network_interface_backend_address_pool_association.my_nic_lb_pool_1 (my_nic_lb_pool_1) will be created
                          + resource "azurerm_network_interface_backend_address_pool_association" "my_nic_lb_pool_1" {
                              + backend_address_pool_id = (known after apply)
                              + id                      = (known after apply)
                              + ip_configuration_name   = "ipconfig-1"
                              + network_interface_id    = (known after apply)
                            }

                          # azurerm_network_security_group.my_nsg (my_nsg) will be created
                          + resource "azurerm_network_security_group" "my_nsg" {
                              + id                  = (known after apply)
                              + location            = "eastus"
                              + name                = "mzemlianyi-azure-cdktf-nsg"
                              + resource_group_name = (known after apply)
                              + security_rule       = [
                                  + {
                                      + access                                     = "Allow"
                                      + description                                = ""
                                      + destination_address_prefix                 = "10.0.1.0/24"
                                      + destination_address_prefixes               = []
                                      + destination_application_security_group_ids = []
                                      + destination_port_range                     = "22"
                                      + destination_port_ranges                    = []
mzemlianyi-azure-stack  + direction                                  = "Inbound"
                                      + name                                       = "ssh"
                                      + priority                                   = 1022
                                      + protocol                                   = "Tcp"
                                      + source_address_prefix                      = "*"
                                      + source_address_prefixes                    = []
                                      + source_application_security_group_ids      = []
                                      + source_port_range                          = "*"
                                      + source_port_ranges                         = []
                                    },
                                  + {
                                      + access                                     = "Allow"
                                      + description                                = ""
                                      + destination_address_prefix                 = "10.0.1.0/24"
                                      + destination_address_prefixes               = []
                                      + destination_application_security_group_ids = []
                                      + destination_port_range                     = "80"
                                      + destination_port_ranges                    = []
                                      + direction                                  = "Inbound"
                                      + name                                       = "web"
                                      + priority                                   = 1080
                                      + protocol                                   = "Tcp"
                                      + source_address_prefix                      = "*"
                                      + source_address_prefixes                    = []
                                      + source_application_security_group_ids      = []
                                      + source_port_range                          = "*"
                                      + source_port_ranges                         = []
                                    },
                                ]
                              + tags                = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                            }

                          # azurerm_public_ip.my_public_ip_0 (my_public_ip_0) will be created
                          + resource "azurerm_public_ip" "my_public_ip_0" {
                              + allocation_method       = "Static"
                              + ddos_protection_mode    = "VirtualNetworkInherited"
                              + fqdn                    = (known after apply)
                              + id                      = (known after apply)
                              + idle_timeout_in_minutes = 4
                              + ip_address              = (known after apply)
                              + ip_version              = "IPv4"
                              + location                = "eastus"
                              + name                    = "mzemlianyi-azure-cdktf-public-ip-0"
                              + resource_group_name     = (known after apply)
                              + sku                     = "Standard"
                              + sku_tier                = "Regional"
                            }

                          # azurerm_public_ip.my_public_ip_1 (my_public_ip_1) will be created
                          + resource "azurerm_public_ip" "my_public_ip_1" {
                              + allocation_method       = "Static"
                              + ddos_protection_mode    = "VirtualNetworkInherited"
                              + fqdn                    = (known after apply)
                              + id                      = (known after apply)
                              + idle_timeout_in_minutes = 4
                              + ip_address              = (known after apply)
                              + ip_version              = "IPv4"
                              + location                = "eastus"
mzemlianyi-azure-stack  + name                    = "mzemlianyi-azure-cdktf-public-ip-1"
                              + resource_group_name     = (known after apply)
                              + sku                     = "Standard"
                              + sku_tier                = "Regional"
                            }

                          # azurerm_resource_group.my_resource_group (my_resource_group) will be created
                          + resource "azurerm_resource_group" "my_resource_group" {
                              + id       = (known after apply)
                              + location = "eastus"
                              + name     = (known after apply)
                              + tags     = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                            }

                          # azurerm_subnet.my_bastion_subnet (my_bastion_subnet) will be created
                          + resource "azurerm_subnet" "my_bastion_subnet" {
                              + address_prefixes                               = [
                                  + "10.0.2.0/24",
                                ]
                              + default_outbound_access_enabled                = true
                              + enforce_private_link_endpoint_network_policies = (known after apply)
                              + enforce_private_link_service_network_policies  = (known after apply)
                              + id                                             = (known after apply)
                              + name                                           = "AzureBastionSubnet"
                              + private_endpoint_network_policies              = (known after apply)
                              + private_endpoint_network_policies_enabled      = (known after apply)
                              + private_link_service_network_policies_enabled  = (known after apply)
                              + resource_group_name                            = (known after apply)
                              + virtual_network_name                           = "mzemlianyi-azure-cdktf-vnet"
                            }

                          # azurerm_subnet.my_subnet (my_subnet) will be created
                          + resource "azurerm_subnet" "my_subnet" {
                              + address_prefixes                               = [
                                  + "10.0.1.0/24",
                                ]
                              + default_outbound_access_enabled                = true
                              + enforce_private_link_endpoint_network_policies = (known after apply)
                              + enforce_private_link_service_network_policies  = (known after apply)
                              + id                                             = (known after apply)
                              + name                                           = "mzemlianyi-azure-cdktf-subnet"
                              + private_endpoint_network_policies              = (known after apply)
                              + private_endpoint_network_policies_enabled      = (known after apply)
                              + private_link_service_network_policies_enabled  = (known after apply)
                              + resource_group_name                            = (known after apply)
                              + virtual_network_name                           = "mzemlianyi-azure-cdktf-vnet"
                            }

                          # azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association (my_nat_gateway_subnet_association) will be created
                          + resource "azurerm_subnet_nat_gateway_association" "my_nat_gateway_subnet_association" {
                              + id             = (known after apply)
                              + nat_gateway_id = (known after apply)
                              + subnet_id      = (known after apply)
                            }

                          # azurerm_subnet_network_security_group_association.my_nsg_association (my_nsg_association) will be created
                          + resource "azurerm_subnet_network_security_group_association" "my_nsg_association" {
                              + id                        = (known after apply)
mzemlianyi-azure-stack  + network_security_group_id = (known after apply)
                              + subnet_id                 = (known after apply)
                            }

                          # azurerm_virtual_machine_extension.my_vm_extension_0 (my_vm_extension_0) will be created
                          + resource "azurerm_virtual_machine_extension" "my_vm_extension_0" {
                              + failure_suppression_enabled = false
                              + id                          = (known after apply)
                              + name                        = "Nginx"
                              + publisher                   = "Microsoft.Azure.Extensions"
                              + settings                    = jsonencode(
                                    {
                                      + commandToExecute = "sudo apt-get update && sudo apt-get install nginx -y && echo \"Greetings from the Azure VM with hostname: $(hostname)\" > /var/www/html/index.html && sudo systemctl restart nginx"
                                    }
                                )
                              + type                        = "CustomScript"
                              + type_handler_version        = "2.0"
                              + virtual_machine_id          = (known after apply)
                            }

                          # azurerm_virtual_machine_extension.my_vm_extension_1 (my_vm_extension_1) will be created
                          + resource "azurerm_virtual_machine_extension" "my_vm_extension_1" {
                              + failure_suppression_enabled = false
                              + id                          = (known after apply)
                              + name                        = "Nginx"
                              + publisher                   = "Microsoft.Azure.Extensions"
                              + settings                    = jsonencode(
                                    {
                                      + commandToExecute = "sudo apt-get update && sudo apt-get install nginx -y && echo \"Greetings from the Azure VM with hostname: $(hostname)\" > /var/www/html/index.html && sudo systemctl restart nginx"
                                    }
                                )
                              + type                        = "CustomScript"
                              + type_handler_version        = "2.0"
                              + virtual_machine_id          = (known after apply)
                            }

                          # azurerm_virtual_network.my_virtual_network (my_virtual_network) will be created
                          + resource "azurerm_virtual_network" "my_virtual_network" {
                              + address_space       = [
                                  + "10.0.0.0/16",
                                ]
                              + dns_servers         = (known after apply)
                              + guid                = (known after apply)
                              + id                  = (known after apply)
                              + location            = "eastus"
                              + name                = "mzemlianyi-azure-cdktf-vnet"
                              + resource_group_name = (known after apply)
                              + subnet              = (known after apply)
                              + tags                = {
                                  + "Course"       = "Introduction to Cloud"
                                  + "Environment"  = "dev"
                                  + "Institution"  = "SET University"
                                  + "ManagedBy"    = "Python CDKTF"
                                  + "Student"      = "mzemlianyi"
                                  + "StudentEmail" = "m.zemlianyi@setuniversity.edu.ua"
                                  + "Task"         = "Practical Task"
                                }
                            }

                          # random_password.random_pass (random_pass) will be created
                          + resource "random_password" "random_pass" {
                              + bcrypt_hash      = (sensitive value)
                              + id               = (known after apply)
                              + length           = 16
                              + lower            = true
                              + min_lower        = 0
                              + min_numeric      = 0
                              + min_special      = 0
                              + min_upper        = 0
                              + number           = true
                              + numeric          = true
                              + override_special = "^!#$%&*()-_=+[]{}<>:?"
                              + result           = (sensitive value)
                              + special          = false
                              + upper            = true
                            }
mzemlianyi-azure-stack    # random_string.my_random_string (my_random_string) will be created
                          + resource "random_string" "my_random_string" {
                              + id          = (known after apply)
                              + length      = 8
                              + lower       = true
                              + min_lower   = 0
                              + min_numeric = 0
                              + min_special = 0
                              + min_upper   = 0
                              + number      = true
                              + numeric     = true
                              + result      = (known after apply)
                              + special     = false
                              + upper       = false
                            }

                        Plan: 28 to add, 0 to change, 0 to destroy.

                        Changes to Outputs:
                          + bastion_generated_password = (known after apply)
                          + bastion_username           = "mzemlianyi"
                          + private_ip_address         = (known after apply)
mzemlianyi-azure-stack  random_string.my_random_string: Creating...
mzemlianyi-azure-stack  random_password.random_pass: Creating...
mzemlianyi-azure-stack  random_string.my_random_string: Creation complete after 0s [id=ofuc5x0h]
mzemlianyi-azure-stack  random_password.random_pass: Creation complete after 0s [id=none]
mzemlianyi-azure-stack  azurerm_resource_group.my_resource_group: Creating...
mzemlianyi-azure-stack  azurerm_resource_group.my_resource_group: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_resource_group.my_resource_group: Creation complete after 12s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h]
mzemlianyi-azure-stack  azurerm_virtual_network.my_virtual_network: Creating...
                        azurerm_public_ip.my_public_ip_1: Creating...
mzemlianyi-azure-stack  azurerm_nat_gateway.my_nat_gateway: Creating...
                        azurerm_public_ip.my_public_ip_0: Creating...
mzemlianyi-azure-stack  azurerm_network_security_group.my_nsg: Creating...
mzemlianyi-azure-stack  azurerm_public_ip.my_public_ip_0: Creation complete after 3s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/publicIPAddresses/mzemlianyi-azure-cdktf-public-ip-0]
mzemlianyi-azure-stack  azurerm_public_ip.my_public_ip_1: Creation complete after 3s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/publicIPAddresses/mzemlianyi-azure-cdktf-public-ip-1]
mzemlianyi-azure-stack  azurerm_network_security_group.my_nsg: Creation complete after 4s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/networkSecurityGroups/mzemlianyi-azure-cdktf-nsg]
mzemlianyi-azure-stack  azurerm_virtual_network.my_virtual_network: Still creating... [10s elapsed]
                        azurerm_nat_gateway.my_nat_gateway: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_nat_gateway.my_nat_gateway: Creation complete after 13s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/natGateways/mzemlianyi-azure-cdktf-nat]
mzemlianyi-azure-stack  azurerm_nat_gateway_public_ip_association.my_nat_gateway_ip_association: Creating...
mzemlianyi-azure-stack  azurerm_virtual_network.my_virtual_network: Creation complete after 17s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/virtualNetworks/mzemlianyi-azure-cdktf-vnet]
mzemlianyi-azure-stack  azurerm_subnet.my_bastion_subnet: Creating...
mzemlianyi-azure-stack  azurerm_subnet.my_subnet: Creating...
mzemlianyi-azure-stack  azurerm_nat_gateway_public_ip_association.my_nat_gateway_ip_association: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_subnet.my_subnet: Creation complete after 7s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/virtualNetworks/mzemlianyi-azure-cdktf-vnet/subnets/mzemlianyi-azure-cdktf-subnet]
mzemlianyi-azure-stack  azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association: Creating...
mzemlianyi-azure-stack  azurerm_subnet_network_security_group_association.my_nsg_association: Creating...
mzemlianyi-azure-stack  azurerm_network_interface.my_nic_0: Creating...
                        azurerm_network_interface.my_nic_1: Creating...
                        azurerm_lb.my_lb: Creating...
mzemlianyi-azure-stack  azurerm_network_interface.my_nic_2: Creating...
mzemlianyi-azure-stack  azurerm_subnet.my_bastion_subnet: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_nat_gateway_public_ip_association.my_nat_gateway_ip_association: Creation complete after 16s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/natGateways/mzemlianyi-azure-cdktf-nat|/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/publicIPAddresses/mzemlianyi-azure-cdktf-public-ip-0]
mzemlianyi-azure-stack  azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association: Still creating... [10s elapsed]
                        azurerm_subnet_network_security_group_association.my_nsg_association: Still creating... [10s elapsed]
                        azurerm_network_interface.my_nic_0: Still creating... [10s elapsed]
                        azurerm_network_interface.my_nic_1: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_lb.my_lb: Still creating... [10s elapsed]
                        azurerm_network_interface.my_nic_2: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_lb.my_lb: Creation complete after 10s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/loadBalancers/mzemlianyi-azure-cdktf-lb]
mzemlianyi-azure-stack  azurerm_lb_backend_address_pool.my_lb_pool: Creating...
mzemlianyi-azure-stack  azurerm_lb_probe.my_lb_probe: Creating...
mzemlianyi-azure-stack  azurerm_subnet.my_bastion_subnet: Still creating... [20s elapsed]
mzemlianyi-azure-stack  azurerm_subnet.my_bastion_subnet: Creation complete after 24s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/virtualNetworks/mzemlianyi-azure-cdktf-vnet/subnets/AzureBastionSubnet]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Creating...
mzemlianyi-azure-stack  azurerm_network_interface.my_nic_0: Still creating... [20s elapsed]
                        azurerm_network_interface.my_nic_2: Still creating... [20s elapsed]
                        azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association: Still creating... [20s elapsed]
                        azurerm_network_interface.my_nic_1: Still creating... [20s elapsed]
                        azurerm_subnet_network_security_group_association.my_nsg_association: Still creating... [20s elapsed]
mzemlianyi-azure-stack  azurerm_lb_probe.my_lb_probe: Still creating... [10s elapsed]
                        azurerm_lb_backend_address_pool.my_lb_pool: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_subnet_network_security_group_association.my_nsg_association: Creation complete after 24s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/virtualNetworks/mzemlianyi-azure-cdktf-vnet/subnets/mzemlianyi-azure-cdktf-subnet]
mzemlianyi-azure-stack  azurerm_lb_probe.my_lb_probe: Creation complete after 14s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/loadBalancers/mzemlianyi-azure-cdktf-lb/probes/test-probe]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_lb_backend_address_pool.my_lb_pool: Creation complete after 18s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/loadBalancers/mzemlianyi-azure-cdktf-lb/backendAddressPools/test-pool]
mzemlianyi-azure-stack  azurerm_lb_rule.my_lb_rule: Creating...
mzemlianyi-azure-stack  azurerm_network_interface.my_nic_2: Still creating... [30s elapsed]
                        azurerm_network_interface.my_nic_0: Still creating... [30s elapsed]
                        azurerm_network_interface.my_nic_1: Still creating... [30s elapsed]
                        azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association: Still creating... [30s elapsed]
mzemlianyi-azure-stack  azurerm_lb_rule.my_lb_rule: Creation complete after 8s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/loadBalancers/mzemlianyi-azure-cdktf-lb/loadBalancingRules/test-rule]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [20s elapsed]
mzemlianyi-azure-stack  azurerm_network_interface.my_nic_1: Creation complete after 40s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/networkInterfaces/mzemlianyi-azure-cdktf-nsg-1]
mzemlianyi-azure-stack  azurerm_network_interface_backend_address_pool_association.my_nic_lb_pool_1: Creating...
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_1: Creating...
mzemlianyi-azure-stack  azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association: Still creating... [40s elapsed]
                        azurerm_network_interface.my_nic_0: Still creating... [40s elapsed]
                        azurerm_network_interface.my_nic_2: Still creating... [40s elapsed]
mzemlianyi-azure-stack  azurerm_network_interface_backend_address_pool_association.my_nic_lb_pool_1: Creation complete after 4s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/networkInterfaces/mzemlianyi-azure-cdktf-nsg-1/ipConfigurations/ipconfig-1|/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/loadBalancers/mzemlianyi-azure-cdktf-lb/backendAddressPools/test-pool]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [30s elapsed]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_1: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association: Still creating... [50s elapsed]
                        azurerm_network_interface.my_nic_2: Still creating... [50s elapsed]
                        azurerm_network_interface.my_nic_0: Still creating... [50s elapsed]
mzemlianyi-azure-stack  azurerm_network_interface.my_nic_2: Creation complete after 53s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/networkInterfaces/mzemlianyi-azure-cdktf-nsg-2]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_2: Creating...
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [40s elapsed]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_1: Still creating... [20s elapsed]
mzemlianyi-azure-stack  azurerm_network_interface.my_nic_0: Still creating... [1m0s elapsed]
                        azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association: Still creating... [1m0s elapsed]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_1: Creation complete after 21s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Compute/virtualMachines/mzemlianyi-azure-cdktf-vm-1]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Creating...
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_2: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_network_interface.my_nic_0: Creation complete after 1m6s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/networkInterfaces/mzemlianyi-azure-cdktf-nsg-0]
mzemlianyi-azure-stack  azurerm_network_interface_backend_address_pool_association.my_nic_lb_pool_0: Creating...
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_0: Creating...
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [50s elapsed]
mzemlianyi-azure-stack  azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association: Still creating... [1m10s elapsed]
mzemlianyi-azure-stack  azurerm_network_interface_backend_address_pool_association.my_nic_lb_pool_0: Creation complete after 4s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/networkInterfaces/mzemlianyi-azure-cdktf-nsg-0/ipConfigurations/ipconfig-0|/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/loadBalancers/mzemlianyi-azure-cdktf-lb/backendAddressPools/test-pool]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_subnet_nat_gateway_association.my_nat_gateway_subnet_association: Creation complete after 1m11s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/virtualNetworks/mzemlianyi-azure-cdktf-vnet/subnets/mzemlianyi-azure-cdktf-subnet]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_2: Creation complete after 19s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Compute/virtualMachines/mzemlianyi-azure-cdktf-vm-2]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_0: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [1m0s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Still creating... [20s elapsed]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_0: Still creating... [20s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [1m10s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Still creating... [30s elapsed]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_0: Still creating... [30s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [1m20s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Still creating... [40s elapsed]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_0: Still creating... [40s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [1m30s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Still creating... [50s elapsed]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_0: Still creating... [50s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [1m40s elapsed]
mzemlianyi-azure-stack  azurerm_linux_virtual_machine.my_vm_0: Creation complete after 53s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Compute/virtualMachines/mzemlianyi-azure-cdktf-vm-0]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_0: Creating...
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Still creating... [1m0s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [1m50s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_0: Still creating... [10s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Still creating... [1m10s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [2m0s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_0: Still creating... [20s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Still creating... [1m20s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [2m10s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_1: Creation complete after 1m27s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Compute/virtualMachines/mzemlianyi-azure-cdktf-vm-1/extensions/Nginx]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_0: Still creating... [30s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [2m20s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_0: Still creating... [40s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [2m30s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_0: Still creating... [50s elapsed]
mzemlianyi-azure-stack  azurerm_virtual_machine_extension.my_vm_extension_0: Creation complete after 56s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Compute/virtualMachines/mzemlianyi-azure-cdktf-vm-0/extensions/Nginx]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [2m40s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [2m50s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [3m0s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [3m10s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [3m20s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [3m30s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [3m40s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [3m50s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [4m0s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [4m10s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [4m20s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [4m30s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [4m40s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [4m50s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [5m0s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [5m10s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [5m20s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [5m30s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [5m40s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [5m50s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [6m0s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [6m10s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [6m20s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [6m30s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [6m40s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [6m50s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [7m0s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [7m10s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [7m20s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [7m30s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [7m40s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [7m50s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Still creating... [8m0s elapsed]
mzemlianyi-azure-stack  azurerm_bastion_host.my_bastion: Creation complete after 8m9s [id=/subscriptions/c265face-b525-4252-97e0-82e82734e7ef/resourceGroups/mzemlianyi-azure-cdktf-public-ip-ofuc5x0h/providers/Microsoft.Network/bastionHosts/mzemlianyi-azure-cdktf-bastion]
mzemlianyi-azure-stack
                        Apply complete! Resources: 28 added, 0 changed, 0 destroyed.
mzemlianyi-azure-stack
                        Outputs:

                        bastion_generated_password = "<omitted_as_sensitive>"
                        bastion_username = "mzemlianyi"
                        private_ip_address = "http://10.0.1.4"

  mzemlianyi-azure-stack
  bastion_generated_password = <omitted_as_sensitive>
  bastion_username = mzemlianyi
  private_ip_address = http://10.0.1.4

RBAC Cleanup triggered
Service Principal with ID "b4aee2f5-c7c9-430d-9944-91af8d96e5c9" exists.
Deleting Azure RBAC credentials with "b4aee2f5-c7c9-430d-9944-91af8d96e5c9"...
Cleanup Azure CLI credentials...
```

LGTM now